R/bind_tibbles.R

```diff
@@ -43,6 +43,10 @@
 bind_tibbles <- function(supertbl,
                          environment = global_env(),
                          tbls = NULL) {
+  check_arg_is_dataframe(supertbl)
```

Comment:
Sounds like a good idea. Not sure we want to check for presence of `redcap_metadata` here since `bind_tibbles()` doesn't need it anyways. Should `check_arg_is_supertbl()` possibly have options to specify which columns are required? Or is an eager check that always requires both columns fine?

Reply:
Is there any way that we can add `supertbl` as on object that gets returned from `class()`? Admittedly I don't know the details of what's required to make something like that happen.

Reply:
Since we're leaning so heavily on the supertibble being a specific thing unique to our package, it could be useful for validations like this, but I'm not sure if it's really the best use case for it.

http://adv-r.had.co.nz/OO-essentials.html#rc

Reply:
Yeah good idea. It looks like we could do it pretty easily with S3 (https://adv-r.hadley.nz/s3.html#s3-classes)

```r
# Would add this at the end of `read_redcap()`
class(supertbl) <- c("our_class", class(supertbl))

# Then check with
inherits(supertbl, "our_class")
```

My biggest question is whether we should do something like this in this PR or think about it more first. We can always implement a simple `check_arg_is_supertbl()` here that doesn't rely on S3 and then replace the internals later.

Reply:
Choosing to add a non-S3 `check_arg_is_supertbl()` for now but we can revisit

Reply:
Yea I think this is something we should implement outside of this PR, and if we want to move forward with it would make for a great addition to 0.3. Documentation could update to talking about how "not only is this a term we're using, but it's a reference-able class as well!"

Reply:
Hm, I think adding a custom S3 class really isn't a big deal and would help identify incorrect inputs so think in scope for this PR. For the check, we might want to make sure that the class is correct and also that critical columns are there (`redcap_data`, `redcap_metadata`, what else?).

Reply:
One issue I'm running into:

In some of our documentation examples we make `supertbl` objects from scratch (see `bind_tibbles()` for example). If we want to keep those examples we'll need either:
1. An exported way to convert `tibbles` to `supertbls`
    - Not so hard to do but we wouldn't want users to actually use this function!
2. **Or** to manually set the S3 class in our examples
    - Works but really ugly documentation

Checks are currently failing because of the examples

Reply:
Actually came up with an **option 3** just after sending:

3. Export a pre-built example `supertbl` that we can use for the examples which already has the right class

I think this is actually the best solution. Thoughts?

Reply:
Option 3 LGTM 👍